src/entity/Ship.java

```diff
@@ -198,12 +198,14 @@ public final int getSpeed() {
 	}
 
 	/**
-	 * Calculates and returns the attack speed in bullets per second.
+	 * Calculates and returns the bullet speed in Pixels per frame.
 	 *
-	 * @return Attack speed (bullets per second).
+	 * @return bullet speed (Pixels per frame).
 	 */
-	public final double getAttackSpeed() {
-		return 1000.0 / growth.getShootingDelay();
+	public final double getBulletSpeed() {
+		double speed = growth.getBulletSpeed();
+		if (speed >= 0) return speed;
+		else return speed * (-1);
```

Comment:
Using Math.abs() makes the code simpler. Like this

return Math.abs(growth.getBulletSpeed());

Reply:
t seems that for the sake of code stability and performance, it would be better not to use the abs() function

Reply:
I think the reviewer's opinion is good.

Reply:
> I think the reviewer's opinion is good.

Noted.

Reply:
Why do you think using the abs() function reduces the stability and performance of your code?

Reply:
There are certain numbers(MIN_VALUE) for which the abs() function does not work correctly.

Reply:
Ah, I see. Thank you for your kind explanation.